app/controllers/admin/products_v3_controller.rb

```diff
@@ -31,6 +31,42 @@ def bulk_update
       end
     end
 
+    def destroy
+      @record = ProductScopeQuery.new(
+        spree_current_user,
+        { id: params[:id] }
+      ).find_product
+
+      status = :ok
+      if @record.destroy
+        flash.now[:success] = I18n.t('admin.products_v3.delete_product.success')
```

Comment:
🤔  I wonder if there's a pattern we can follow here. 
My first thought is to use [Lazy Lookup](https://guides.rubyonrails.org/i18n.html#lazy-lookup) like we do in views. 
Then I looked and found we already have an embedded pattern in the `ResourceController`, which seems appropriate. But it's not very flexible and I'm unsure if we want to keep using that.

So I don't know if we should change it..

Reply:
in most of the places in controllers, we use `I18n.t` while in views we use `t`. I just learned that we can use `t` in controllers as well. Thanks for sharing.
It's fixed here:
https://github.com/openfoodfoundation/openfoodnetwork/commit/ce60a2a1e0fddca0781d4894a6c76fb3c9a3d486